build-tools/packages/build-cli/src/commands/generate/entrypoints.ts

```diff
@@ -0,0 +1,191 @@
+// eslint-disable-next-line import/no-internal-modules
+import { ApiLevel } from "../../library/apiLevel.js";
+// eslint-disable-next-line import/no-internal-modules
+import { getApiExports } from "../../library/typescriptApi.js";
+import type { CommandLogger } from "../../logging.js";
+
+/**
+ * Literal pattern to search for in file prefix to replace with unscoped package name.
+ *
+ * @privateRemarks api-extractor use `<@..>`, but `<>` is problematic for command line
+ * specification. A policy incorrectly thinks an argument like that should not be quoted.
+ * It is just easier to use an alternate bracket style.
+ */
+const unscopedPackageNameString = "{@unscopedPackageName}";
+
+/**
+ * Generates type declarations files for Fluid Framework APIs to support API levels (/alpha, /beta. etc.).
+ */
+export default class GenerateEntrypointsCommand extends BaseCommand<
+	typeof GenerateEntrypointsCommand
+> {
+	static readonly description =
+		`Generates type declaration entrypoints for Fluid Framework API levels (/alpha, /beta. etc.)`;
+
+	static readonly flags = {
+		mainEntrypoint: Flags.file({
+			description: "Main entrypoint file containing all untrimmed exports.",
+			default: "./src/index.ts",
+			exists: true,
+		}),
+		outDir: Flags.directory({
+			description: "Directory to emit entrypoint declaration files.",
+			default: "./lib",
+			exists: true,
+		}),
+		outFilePrefix: Flags.string({
+			description: `File name prefix for emitting entrypoint declaration files. Pattern of '${unscopedPackageNameString}' within value will be replaced with the unscoped name of this package.`,
+			default: "",
+		}),
+		outFileSuffix: Flags.string({
+			description:
+				"File name suffix including extension for emitting entrypoint declaration files.",
+			default: ".d.ts",
+		}),
+		...BaseCommand.flags,
+	};
+
+	public async run(): Promise<void> {
+		const { mainEntrypoint, outFileSuffix } = this.flags;
+
+		return generateEntrypoints(
+			mainEntrypoint,
+			{ pathPrefix: await getOutPathPrefix(this.flags), pathSuffix: outFileSuffix },
+			this.logger,
+		);
+	}
+}
+
+async function getOutPathPrefix({
+	outDir,
+	outFilePrefix,
+}: { outDir: string; outFilePrefix: string }): Promise<string> {
+	if (!outFilePrefix) {
+		// If no other prefix, ensure a trailing path separator.
+		// The join with '.' will effectively trim a trailing / or \ from outDir.
+		return `${path.join(outDir, ".")}${path.sep}`;
+	}
+
+	return path.join(
+		outDir,
+		outFilePrefix.includes(unscopedPackageNameString)
+			? outFilePrefix.replace(unscopedPackageNameString, await getLocalUnscopedPackageName())
+			: outFilePrefix,
+	);
+}
+
+async function getLocalUnscopedPackageName(): Promise<string> {
+	const packageJson = await fs.readFile("./package.json", { encoding: "utf8" });
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
+	const packageName = JSON.parse(packageJson).name;
```

Comment:
Use the PackageJson type from build-tools.

	const packageName = (JSON.parse(packageJson) as PackageJson).name;

```suggestion
	const packageName = (JSON.parse(packageJson) as PackageJson).name;
```